File: src/main/java/io/gitlab/intended/storagenetworks/registry/ItemRegistry.java
package io.gitlab.intended.storagenetworks.registry;

import io.gitlab.intended.storagenetworks.StorageNetworks;
import net.minecraft.item.Item;
import net.minecraft.util.registry.Registry;

public final class ItemRegistry {

    public static void register() {
        
    }

    protected static void register(String id, Item item) {
        Registry.register(Registry.ITEM, StorageNetworks.id(id), item);
    }

}

File: src/main/kotlin/badasintended/slotlink/item/ItemRegistry.kt
@file:Suppress("MemberVisibilityCanBePrivate")

package badasintended.slotlink.item

import net.minecraft.util.registry.Registry
import net.minecraft.util.registry.Registry.ITEM

object ItemRegistry {

    val UNLIMITED_REMOTE = UnlimitedRemoteItem()
    val LIMITED_REMOTE = LimitedRemoteItem()

    fun init() {
        r(UNLIMITED_REMOTE, LIMITED_REMOTE)
    }

    private fun r(vararg items: ModItem) {
        items.forEach { Registry.register(ITEM, it.id, it) }
    }

}
